src/main/scala/com/fulcrumgenomics/math/BinomialDistribution.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2017 Fulcrum Genomics LLC
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.fulcrumgenomics.math
+
+import com.fulcrumgenomics.FgBioDef._
+
+import scala.math.{BigDecimal, BigInt}
+
+/**
+  * A high-precision implementation of the math for binomial probabilities.  This implementation is several
+  * times slower (though not orders of magnitude slower) than the implementation in Apache Commons Math, but
+  * retains precision throughout the computation where the Commons implementation underflows.
+  *
+  * One implementation choice to be aware of when using this implementation is that each instance of the
+  * class will calculate and cache factorials up to factorial(n) where n is the highest value of n
+  * supplied to any call to [[probability()]] or [[cumulativeProbability()]].  Thus to ensure reasonable
+  * performance it is recommended to create one instance of this class and reuse it for as may
+  * calculations as is practical.
+  */
+class BinomialDistribution {
+  // This array will get expanded with more factorials any time a higher n is queried
+  private var factorials = Array(BigInt(0), BigInt(1))
+
+  // Constant values for 0 and 1 in BigDecimal to avoid making them all the time
+  private val Zero = BigDecimal(0)
```

Comment:
BigDecimal already has a static ONE and ZERO.

Reply:
For some reason, the scala wrapper of BigDecimal doesn't expose these, and it's a different (i.e. wrapper) type than java.math.BigDecimal.